pre_challenge/lib/state/provider.dart

```diff
@@ -1,9 +1,34 @@
-import 'package:riverpod/riverpod.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'import.dart';
 
 //ラベルが格納されているStateProvider
-final StateProvider<List> labelProvider = StateProvider((ref) => ["全て","p: webview","p: shared_preferences","waiting for customer response","sever: new feature","p: share"]);
 
-//以下GraphQLを利用する必要があるProvider
+class labelNotifier extends Notifier<List> {
+@override
+  List build() => [
+    "全て",
+    "p: webview",
+    "p: shared_preferences",
+    "waiting for customer response",
+    "sever: new feature",
+    "p: share"
+  ];
+
+  void addLabel(String label) {
+    state.add(label);
```

Comment:
ドロワーに即時反映されないの、この行の書き方が原因です
```suggestion
    state = state[...state, label];
```

Notifierの動作として
- stateを保持する
- state が 変わったら、ref.watch しているWidgetを再ビルドする

という物があるんですが、この書き方では「stateが変わったかどうか」を正しく判定できません。addする前のリストもaddしたあとのリストも同じリストと判定され、再ビルドが発火しません

なぜ？ってのはさておくと（数行で説明するのむずい）、次のように修正すれば再ビルドされるようになると思うので試してみてください

Reply:
修正しました。Notifierの動作については後ほど学習します。